File: src/main/java/cn/afeibaili/mchat/logger/Logger.java
package cn.afeibaili.mchat.logger;


import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * 日志类
 *
 * @author AfeiBaili
 * @version 2026/3/3 00:12
 */

public interface Logger {
    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss.SSS");

    ExecutorService executor = Executors.newSingleThreadExecutor();

    void setName(String name);

    String getName();

    void info(String message);

    void warn(String message);

    void error(String message);

    default void print(String level, String message) {
        executor.execute(() -> {
            System.out.println("[" + level + "] " + getDataTime() + ": " + message);
        });
    }

    default String getDataTime() {
        return LocalDateTime.now().format(formatter);
    }

    static Logger getLogger() {
        return new Logger() {
            String name;

            @Override
            public void setName(String name) {
                this.name = name;
            }

            @Override
            public String getName() {
                return name;
            }

            @Override
            public void info(String message) {
                print("INFO", message);
            }

            @Override
            public void warn(String message) {
                print("WARN", message);
            }

            @Override
            public void error(String message) {
                print("ERROR", message);
            }
        };
    }
}

File: core/src/main/kotlin/online/afeibaili/mchat/logger/Logger.kt
package online.afeibaili.mchat.logger

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter


/**
 * 日志器
 *
 *@author AfeiBaili
 *@version 2026/2/10 01:18
 */

interface Logger {
    val name: String
    fun info(message: Any)
    fun error(message: Any)
    fun warn(message: Any)
    fun print(level: String, message: Any) {
        scope.launch {
            println("[$level] ${getDataTime()}: $message")
        }
    }

    private fun getDataTime() = LocalDateTime.now().format(formatter)

    companion object {
        val scope = CoroutineScope(Dispatchers.Default)
        private val formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss.SSS")
        fun getLogger(name: Any): Logger {
            return object : Logger {
                override val name: String = name.toString()

                override fun info(message: Any) {
                    print("INFO", message)
                }

                override fun error(message: Any) {
                    print("ERROR", message)
                }

                override fun warn(message: Any) {
                    print("WARN", message)
                }
            }
        }
    }
}